contracts/CollectiveGovernance.sol

```diff
@@ -552,13 +635,68 @@ contract CollectiveGovernance is Governance, VoteStrategy, ERC165 {
         return _communityDescription;
     }
 
+    function castVoteFor(uint256 _proposalId, uint256 _tokenId) private {
+        uint256 count = _storage.voteForByShare(_proposalId, msg.sender, _tokenId);
+        if (count > 0) {
+            emit VoteCount(_proposalId, msg.sender, _tokenId, count, 0);
+        } else {
+            revert("Not voter");
+        }
+    }
+
+    function castVoteAgainst(uint256 _proposalId, uint256 _tokenId) private {
+        uint256 count = _storage.voteAgainstByShare(_proposalId, msg.sender, _tokenId);
+        if (count > 0) {
+            emit VoteCount(_proposalId, msg.sender, _tokenId, 0, count);
+        } else {
+            revert("Not voter");
+        }
+    }
+
+    function castAbstention(uint256 _proposalId, uint256 _tokenId) private {
+        uint256 count = _storage.abstainForShare(_proposalId, msg.sender, _tokenId);
+        if (count > 0) {
+            emit VoteCount(_proposalId, msg.sender, _tokenId, 0, 0);
+        } else {
+            revert("Not voter");
+        }
+    }
+
+    function _undoVote(uint256 _proposalId, uint256 _tokenId) private {
+        uint256 count = _storage.undoVoteById(_proposalId, msg.sender, _tokenId);
+        if (count > 0) {
+            emit VoteUndo(_proposalId, msg.sender, count);
+        } else {
+            revert("Not voter");
+        }
+    }
+
+    function sendGasRebate(address recipient, uint256 startGas) internal {
+        uint256 balance = getRebateBalance();
+        if (balance == 0) {
+            return;
+        }
+
+        uint256 totalGasUsed = startGas - gasleft();
```

Comment:
should we move this down two lines to maximize the gas refunded to user? otherwise they have to pay for those two lines

Reply:
I could go either way but it seems like a trivial maybe even not measurable change to the gas used, while on the other hand seemingly lowering the readability and understandability of this code.    

> should we move this down two lines to maximize the gas refunded to user? otherwise they have to pay for those two lines